### PR TITLE
weston: 11.0.0 -> 11.0.1

### PR DESCRIPTION
**Pull request by wineee**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/weston/default.nix [hunk 1]
```diff
@@ -24,20 +28,21 @@ stdenv.mkDerivation rec {
...
     "-Dxwayland=${lib.boolToString (xwayland != null)}" # Default is true!
-    "-Dremoting=false" # TODO
+    (lib.mesonBool "remoting" buildRemoting)
     "-Dpipewire=${lib.boolToString (pipewire != null)}"
     "-Dimage-webp=${lib.boolToString (libwebp != null)}"
     (lib.mesonBool "demo-clients" buildDemo)
     "-Dsimple-clients="
     "-Dtest-junit-xml=false"
-    # TODO:
-    #"--enable-clients"
```
Comment (wineee) [inline pkgs/applications/window-managers/weston/default.nix:40]: This option can't be found now

File: pkgs/applications/window-managers/weston/default.nix [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 , meson, ninja, pkg-config, python3, wayland-scanner
 , cairo, colord, dbus, lcms2, libGL, libXcursor, libdrm, libevdev, libinput
 , libjpeg, seatd, libxcb, libxkbcommon, mesa, mtdev, pam, udev, wayland
 , wayland-protocols
 , pipewire ? null, pango ? null, libunwind ? null, freerdp ? null, vaapi ? null
 , libva ? null, libwebp ? null, xwayland ? null
 # beware of null defaults, as the parameters *are* supplied by callPackage by default
-, buildDemo ? false
+, buildDemo ? true
+, buildRemoting ? true, gst_all_1
```
Comment (wineee) [inline pkgs/applications/window-managers/weston/default.nix:10]: Upstream  enabled build demo and remoting by defaul

File: pkgs/applications/window-managers/weston/default.nix [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
 
 stdenv.mkDerivation rec {
   pname = "weston";
-  version = "11.0.0";
+  version = "11.0.1";
 
-  src = fetchurl {
-    url = "https://gitlab.freedesktop.org/wayland/weston/-/releases/${version}/downloads/weston-${version}.tar.xz";
-    sha256 = "078y14ff9wmmbzq314f7bq1bxx0rc12xy4j362n60iamr56qs4x6";
+  src = fetchFromGitLab {
```
Comment (alyssais) [inline pkgs/applications/window-managers/weston/default.nix:17]: Let's keep using the official tarballs unless there's a compelling reason to change.

File: pkgs/applications/window-managers/weston/default.nix (hunk 2 above)
Comment (alyssais) [reply pkgs/applications/window-managers/weston/default.nix:10]: How about turning these off for `westonLite`, and having them on for `weston`?

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/applications/window-managers/weston/default.nix (hunk 3 above)
Comment (alyssais) [reply pkgs/applications/window-managers/weston/default.nix:17]: Yep, that looks right — it's the URL in the release announcement.

**Comment by alyssais** [comment]
@OfBorg build weston westonLite

**Comment by wineee** [comment]
We need backport with: https://github.com/NixOS/nixpkgs/pull/220425/files

**Pull request merged**